src/valuedlg.cpp

```diff
@@ -252,10 +252,19 @@ void ValueDlg::VisualizeQuery()
     m_view->setAttribute(Qt::WA_DeleteOnClose);
 
     // Set Qt::Dialog rather than Qt::Window to disable zoom and avoid a Qt bug with unresponsive zoomed views on OSX
-    m_view->setWindowFlags(Qt::Dialog);
-    m_view->resize(1000, 1000);
+    m_view->setWindowFlags(Qt::Dialog | Qt::WindowMaximizeButtonHint | Qt::WindowCloseButtonHint);
     m_view->setWindowTitle(QString("Query Visualization - ID: %1 - Key: %2").arg(m_id, m_key));
 
+    QDesktopWidget widget;
+    int screenId = widget.screenNumber(this);
+    QRect screenRect = widget.availableGeometry(screenId);
+    int width = screenRect.width() - 400;
```

Comment:
I personally dislike when windows spawn in not-relative-to-parent position but if this flow works for you - I'm fine with that.